pkg/parser/utils.go

```diff
@@ -0,0 +1,281 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All the helper functions should be hosted by another public repo later. (e.g. agentbaker)
+// Helper functions in this file will be called by bootstrappers to populate nb contract payload.
+package parser
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/agentbaker/pkg/agent"
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	nbcontractv1 "github.com/Azure/agentbaker/pkg/proto/nbcontract/v1"
+	"github.com/blang/semver"
+)
+
+// GetLoadBalancerSKI returns the LoadBalancerSku enum based on the input string.
+func GetLoadBalancerSKU(sku string) nbcontractv1.LoadBalancerConfig_LoadBalancerSku {
```

Comment:
Should we host this function in AgentBaker and export it so that client can use it? For example, Karpenter will use it to populate the nbconfig object.

Reply:
This is in agentbaker... and it is exported. But we can move it to a different package if that's what you're suggesting? I also wasn't sure where best to put it. I was thinking maybe the directory anuj created called `selfcontained`.

Reply:
It's in the parser folder so I thought this will be in the parser go binary. pkg/proto/nbcontract/v1/helper.go should be good because client just needs to import single package to have both nbcontract and corresponding helpers.

Reply:
ah okay agreed, moved.